pyscf_ipu/nanoDFT/nanoDFT.py

```diff
@@ -132,13 +137,14 @@ def _nanoDFT(state, ERI, grid_AO, grid_weights, opts, mol):
 
     # Log matrices from all DFT iterations (not used by DFT algorithm).
     N = H_core.shape[0]
-    log = {"matrices": np.zeros((opts.its, 4, N, N)), "E_xc": np.zeros((opts.its)), "energy": np.zeros((opts.its, 5))}
+    log = {"matrices": np.zeros((opts.its, 4, N, N)), "E_xc": np.zeros((opts.its)), "energy": np.zeros((opts.its, 5)), "eigenvectors" : np.zeros((opts.its, N, N))}
```

Comment:
Assuming this is mainly for plotting. Could you try to profile `at.[i].set(..)` compared to `jax.lax.dynamic_update_slice(..)`? I've had issues with the `at[i].set(..)` being 10x slower due to unfortunate memory layout.